python/triton/language/semantic.py

```diff
@@ -1343,10 +1343,18 @@ def dot(lhs: tl.tensor,
         builder: ir.builder) -> tl.tensor:
     def assert_dtypes_valid(lhs_dtype, rhs_dtype, target):
         # Checks for non-cuda archs
-        if not _is_cuda(target):
+        if is_hip():
             assert lhs.dtype == rhs.dtype or (lhs.type.scalar.is_fp8() and rhs.type.scalar.is_fp16()) or \
                 (lhs.type.scalar.is_fp16() and rhs.type.scalar.is_fp8()) or (lhs.type.scalar.is_fp8() and rhs.type.scalar.is_fp8()), \
                 f"First input ({lhs.dtype}) and second input ({rhs.dtype}) must have the same dtype!"
+            if lhs.type.scalar.is_fp8() and rhs.type.scalar.is_fp8():
```

Comment:
For my knowledge, why is it using `lhs.type.scalar.is_fp8()`? Thought is a vector instead of scalar?

Reply:
I think we care only the element type in the vector, right?

Reply:
Yeah. I was just curious why the checking function is in the module called `scalar`.